src/AgreementStructs.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+enum PositionStatus {
+    Idle,
+    Terminated
+}
+
+/// @dev Agreement party position
+struct Position {
+    /// @dev Matches index of the party in the agreement
+    uint256 id;
+    /// @dev Amount of tokens corresponding to this position.
+    uint256 balance;
+    /// @dev Status of the position
+    PositionStatus status;
+}
+
+struct Agreement {
+    /// @dev Hash of the detailed terms of the agreement.
+    bytes32 termsHash;
+    /// @dev Required amount to join or merkle root of (address,amount).
+    uint256 criteria;
+    /// @dev Number of termination confirmations.
+    uint256 terminations;
+    /// @dev List of parties involved in the agreement.
+    address[] party;
+    /// @dev Position by party.
+    mapping(address => Position) position;
```

Comment:
Should this be plural?  `¯\_(ツ)_/¯`
```suggestion
    mapping(address => Position) positions;
```

Reply:
Usually I would do it this way. But while doing research on other implementations I liked how singular noums on mappings and arrays was used. I guess it is that way because when you retrieve an element you do `Agreement.position[address]`. This way of naming is also coherent with the use of `balanceOf` `ownerOf` and other mappings you find in other implementations.

Reply:
Said that, I don't have an especially hard opinion on this so I'm open to change it.